### PR TITLE
New geoms

### PR DESCRIPTION
**Pull request by zmeers**
I think we can push this to the master branch, pending your okay.
It's a lot cleaner than the master branch at this point and has several new functions. 
Just a quick note about the helper functions - we'll want to remove the export, won't we? No need for users to have access to it..

### CODE REVIEW

File: R/draw_legislaturelabels.R
Comment (RobWHickman) [inline R/draw_legislaturelabels.R:11]: this is obviously just an aesthetic change, but I wonder if we should standardise function names. I would propose using geom_ for anything that's a shape/coloured, label_ for these funcs and so on? Will leave unchanged for now

File: R/draw_legislaturelabels.R
Comment (RobWHickman) [inline R/draw_legislaturelabels.R]: at some point we might want to clear up as much of the dplyr/magrittr as possible. Again, not important now. I believe aggregate is an equiv of group_by so I'll play around tomorrow

Comment (zmeers) [reply R/draw_legislaturelabels.R]: Yeah, sure thing. Or play around with enquo and all that jazz. I think labels need to be sorted out so that the user can define all the column names that will be then put into new_dat. We can do this in dplyr pretty easily using enquo but if you prefer base R, that's fine as well.

File: TODO.Rmd
Comment (RobWHickman) [inline TODO.Rmd:49]: this was actually the reason I took the project back up a few months ago and just never got round to it. Definitely possible. Faceting itself isn't that bad. I was trying to play around with faceting+animation that gets tricky with the parliament_data equivalent 👍

File: TODO.Rmd
Comment (RobWHickman) [inline TODO.Rmd:37]: tbh, should only take an hour or so for the recent UK data. Theyworkforyou should have everything

File: TODO.Rmd
Comment (RobWHickman) [inline TODO.Rmd]: 👍 on the NYC tweet thing it looks really good, though I might be tempted to put the party colours underneath? I know in the old version it could mess up the scaling to have the legend on the side.

Comment (zmeers) [reply TODO.Rmd]: Hmm, put the black part in the center? Let's try it and see what it looks like! Just switch around the layers -- new_layers plots first in this case.

Re: theme... I think we should take a look at https://github.com/hrbrmstr/hrbrthemes. There are a few interesting things here. I like that the user can just write TRUE/FALSE for legends or axes or whatever instead of `theme(legend.position='none')`. So if someone wants to draw party labels instead of using a legend, they can write `theme_ggparliament(legend=FALSE)`  or vice-versa if they want the legend. 

Moreover, if someone writes `theme_ggparliament(legend=TRUE)` we can set the legend so that it plots on the bottom. That may screw up the opposing bench style, though. Do you think we should we flip it around so that there is a bench on top and one below? Or I suppose it can switch the type, then plot to the side... 

My point is: I think we should have an opinionated theme, otherwise the user can just create their own.

File: TODO.Rmd
Comment (zmeers) [reply TODO.Rmd:49]: Faceting by year or house is easy using map inside the dplyr chain. i.e.
```
data <- election_data %>%
filter(country == "USA & year == "2016") %>%
  split(.$house) %>% # split
  map(~parliament_data(election_data = ., # apply
  total_seats = sum(.$seats),
  party_seats = .$seats,
  parl_rows = 8,
  type = "semicircle")) %>%
  bind_rows() # combine
```
And then when you plot the figure, just use facet_grid or facet_wrap on the variable that you split the data by. 
I'm really interested in trying to split by country and then plot separate parliament styles.

File: DESCRIPTION
Comment (RobWHickman) [inline DESCRIPTION:9]: not sure what the convention is here

File: R/draw_legislaturelabels.R
Comment (RobWHickman) [reply R/draw_legislaturelabels.R]: I prefer dplyr but have always been taught (albeit informally) that base R is preferred for packages? Happy to use dplyr if that isn't the case though

File: R/draw_majorityline.R
Comment (RobWHickman) [inline R/draw_majorityline.R]: see draw_legislaturelabels. Will probably simplify the ifelse chains using switch calls where possible. Will pull this across before I do that because I don't want to spend ages doing small edits here and there when the branch works but think there's a lot of scope to reduce some of the functions.

Is there a way to pipe the type through ggplot (e.g. p <- ggplot(data = df, aes(type = type)) + draw_majoritythreshold() ) or does it need to be defined in every extra geom/scale/... ?

File: README.md
Comment (RobWHickman) [inline README.md:247]: I think the README looks _really_ great :D 

as always, no idea of the field, but I can imagine theres a lot of use cases for people who are just picking up R for the first/second/... time so I think once we get the code sorted spending some time on READMEs/vignettes/etc. will pay off multiple times.

It might have died down by then but I think at least for the UK, I'd be interested in making a few blog posts wrt the brexit vote chaos in the UK at the moment. Try and pump some use cases out and get pol sci/ journalism people alerted to the package on twitter

**Comment by RobWHickman** [comment]
cool- gone through all the code changes and I'll probs change some small things and try and simplify the code down a bit where theres some redundancy but I don't think theres anything that should hold up a pull.

Going to spend tomorrow lunch/evening going through the vignettes to do a final check and then I'll pull across

File: DESCRIPTION
Comment (zmeers) [reply DESCRIPTION:9]: I guess you go first as the maintainer ("cre"), I'll go second, then Thomas. That way it's in alphabetical order, as well...

File: R/draw_legislaturelabels.R
Comment (zmeers) [reply R/draw_legislaturelabels.R]: I think dplyr itself isn't a good idea because is uses Non Standard Evaluation but rlang, the language that supports dplyr, should be fine. This might be helpful: https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html

File: R/draw_majorityline.R
Comment (zmeers) [reply R/draw_majorityline.R]: Working on piping the type through the aes now. It works for everything but draw_partylabels atm. Will push when I've debugged that function. Exciting stuff :)

**Comment by zmeers** [comment]
Hey -- if you can double check the aes(type=type) stuff, that would be awesome! One issue I've spotted is if you don't load `+draw_majoritylabel()` or whatever a warning pops up sayng that the type is not defined (because it's not needed for `geom_parliament_seats()`. Thanks.

File: vignettes/basic-parliament-plots.Rmd
Comment (RobWHickman) [inline vignettes/basic-parliament-plots.Rmd]: (I don't know what's best practice for this) is it worth explicitly loading libraries that are needed for the munging (magrittr+dplyr / tidyverse/ etc.) or letting the user load them themselves?

**Comment by RobWHickman** [comment]
I was having trouble building the package which was solved by removing collate from the DESCRIPTION file as per https://github.com/klutometis/roxygen/issues/485. Seems like everything is still working but just fyi

(also added the data name to data.R to allow loading of election_data easily)

**Comment by RobWHickman** [comment]
ok, gone through the vignettes, made a few changes:

- put numbers before file names to suggest reading order
- tried to simplify the names and standardise between files/chunks when stuff is named/what its named etc. Probs needs  a bit more cleaning still
- draw_majority_line doesn't seem to be working for me. Says some argument x is missing. I'll take a closer look tomorrow

**Review by RobWHickman — APPROVED** [review]
approved for merge.

**Pull request merged**
